### DISCUSSION TITLE
Suddenly tunnel was down and the logs are pasted below

### DISCUSSION THREAD

**Discussion by harsh-madhav** [strongswan/strongswan#2345] [General]
deleting IKE_SA linode-to-azure[188] between 192...[172...]...20..[20...]
deleting policy 192.168.10.0/24 === 177.30.0.0/16 out
Jul 11 04:53:03 prod-nginx ipsec[675]: 00[KNL] deleting policy 177../16 === 192.../24 in
Jul 11 04:53:03 prod-nginx ipsec[675]: 00[KNL] deleting policy 177.**./16 === 192.../24 fwd
Jul 11 04:53:03 prod-nginx ipsec[675]: 00[KNL] deleting SAD entry with SPI cd992431
Jul 11 04:53:03 prod-nginx ipsec[675]: 00[KNL] deleting SAD entry with SPI 5bbf3f06
please help me with this, what could be the reasons for this?

**Comment by tobiasbrunner** [comment]
This log is way too short to tell anything from it. It's also after the fact (the delete itself is not interesting, why/how there was a delete is). So please provide more logs (if possible also from the other end). See #196.

**Comment by harsh-madhav** [comment]
sure! adding few more logs
Jul 11 00:08:22 prod-nginx charon-custom: 06[IKE] received DELETE for ESP CHILD_SA with SPI 3462e6c4
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]  448: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00  ................
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]  464: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00  ................
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]  480: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00  ................
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]  496: 00 01 00 00 F4 49 42 F9 38 E1 9D 1A 2B C1 1B 22  .....IB.8...+.."
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]  512: 15 10 CC 98 8F D6 EE 78 7D 10 72 6B 46 30 56 A5  .......x}.rkF0V.
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]  528: 35 B7 28 49 1C 00 04 00 02 00 11 94 11 94 00 00  5.(I............
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]  544: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00  ................
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]  560: 10 00 0A 00 42 93 00 00 00 00 00 00 00 00 00 00  ....B...........
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[IKE] closing CHILD_SA linode-to-azure{1036} with SPIs c9c89946_i (15518192 bytes) 3462e6c4_o (25612844 bytes) and TS 192.168.10.0/24 === 178.30.0.0/16
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[IKE] sending DELETE for ESP CHILD_SA with SPI c9c89946
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[IKE] CHILD_SA closed
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL] deleting policy 192.168.10.0/24 === 178.30.0.0/16 out
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL] sending RTM_DELROUTE 1430: => 36 bytes @ 0x7f163028ee40
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]    0: 24 00 00 00 19 00 05 00 96 05 00 00 A3 02 00 00  $...............
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   16: 02 10 00 00 DC 04 00 01 00 00 00 00 08 00 01 00  ................
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   32: B2 1E 00 00                                      ....
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL] received (2) 1430: => 36 bytes @ 0x7f15f0013b00
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]    0: 24 00 00 00 02 00 00 01 96 05 00 00 A3 02 00 00  $...............
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   16: 00 00 00 00 24 00 00 00 19 00 05 00 96 05 00 00  ....$...........
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   32: A3 02 00 00                                      ....
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL] sending XFRM_MSG_DELPOLICY 428291: => 80 bytes @ 0x7f163028f4d0
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]    0: 50 00 00 00 14 00 05 00 03 89 06 00 A3 02 00 00  P...............
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   16: B2 1E 00 00 00 00 00 00 00 00 00 00 00 00 00 00  ................
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   32: C0 A8 0A 00 00 00 00 00 00 00 00 00 00 00 00 00  ................
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   48: 00 00 00 00 00 00 00 00 02 00 10 18 00 00 00 00  ................
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   64: 00 00 00 00 00 00 00 00 00 00 00 00 01 00 00 00  ................
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL] received (2) 428291: => 36 bytes @ 0x7f15f8003a90
Jul 11 00:08:22 prod-nginx charon-custom: 06[IKE] closing CHILD_SA linode-to-azure{1036} with SPIs c9c89946_i (15518192 bytes) 3462e6c4_o (25612844 bytes) and TS 192.168.10.0/24 === 178.30.0.0/16
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]    0: 24 00 00 00 02 00 00 01 03 89 06 00 A3 02 00 00  $...............
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   16: 00 00 00 00 50 00 00 00 14 00 05 00 03 89 06 00  ....P...........
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   32: A3 02 00 00                                      ....
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL] deleting policy 178.30.0.0/16 === 192.168.10.0/24 in
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL] sending XFRM_MSG_DELPOLICY 428292: => 80 bytes @ 0x7f163028f4f0
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]    0: 50 00 00 00 14 00 05 00 04 89 06 00 A3 02 00 00  P...............
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   16: C0 A8 0A 00 00 00 00 00 00 00 00 00 00 00 00 00  ................
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   32: B2 1E 00 00 00 00 00 00 00 00 00 00 00 00 00 00  ................
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   48: 00 00 00 00 00 00 00 00 02 00 18 10 00 00 00 00  ................
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   64: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00  ................
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL] received (2) 428292: => 36 bytes @ 0x7f15f0013b00
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]    0: 24 00 00 00 02 00 00 01 04 89 06 00 A3 02 00 00  $...............
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   16: 00 00 00 00 50 00 00 00 14 00 05 00 04 89 06 00  ....P...........
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   32: A3 02 00 00                                      ....
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL] deleting policy 178.30.0.0/16 === 192.168.10.0/24 fwd
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL] sending XFRM_MSG_DELPOLICY 428293: => 80 bytes @ 0x7f163028f4f0
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]    0: 50 00 00 00 14 00 05 00 05 89 06 00 A3 02 00 00  P...............
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   16: C0 A8 0A 00 00 00 00 00 00 00 00 00 00 00 00 00  ................
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   32: B2 1E 00 00 00 00 00 00 00 00 00 00 00 00 00 00  ................
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   48: 00 00 00 00 00 00 00 00 02 00 18 10 00 00 00 00  ................
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   64: 00 00 00 00 00 00 00 00 00 00 00 00 02 00 00 00  ................
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL] received (2) 428293: => 36 bytes @ 0x7f15f8003a90
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]    0: 24 00 00 00 02 00 00 01 05 89 06 00 A3 02 00 00  $...............
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   16: 00 00 00 00 50 00 00 00 14 00 05 00 05 89 06 00  ....P...........
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   32: A3 02 00 00                                      ....
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL] deleting SAD entry with SPI c9c89946
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL] sending XFRM_MSG_DELSA 428294: => 40 bytes @ 0x7f163028f5a0
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]    0: 28 00 00 00 11 00 05 00 06 89 06 00 A3 02 00 00  (...............
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   16: C0 A8 0A 02 00 00 00 00 00 00 00 00 00 00 00 00  ................
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   32: C9 C8 99 46 02 00 32 00                          ...F..2.
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL] received (2) 428294: => 36 bytes @ 0x7f15ec00a1d0
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]    0: 24 00 00 00 02 00 00 01 06 89 06 00 A3 02 00 00  $...............
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   16: 00 00 00 00 28 00 00 00 11 00 05 00 06 89 06 00  ....(...........
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   32: A3 02 00 00                                      ....
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL] deleted SAD entry with SPI c9c89946
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL] deleting SAD entry with SPI 3462e6c4
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL] sending XFRM_MSG_DELSA 428295: => 40 bytes @ 0x7f163028f5a0
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]    0: 28 00 00 00 11 00 05 00 07 89 06 00 A3 02 00 00  (...............
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   16: 14 31 15 52 00 00 00 00 00 00 00 00 00 00 00 00  .1.R............
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   32: 34 62 E6 C4 02 00 32 00                          4b....2.
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL] received (2) 428295: => 36 bytes @ 0x7f15f80010e0
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]    0: 24 00 00 00 02 00 00 01 07 89 06 00 A3 02 00 00  $...............
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   16: 00 00 00 00 28 00 00 00 11 00 05 00 07 89 06 00  ....(...........
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL]   32: A3 02 00 00                                      ....
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[KNL] deleted SAD entry with SPI 3462e6c4
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[ENC] generating INFORMATIONAL response 0 [ D ]
Jul 11 00:08:22 prod-nginx ipsec[675]: 06[NET] sending packet: from 192.*.*.*[4500] to 20.*.*.*[4500] (76 bytes)
Jul 11 00:08:22 prod-nginx charon-custom: 06[IKE] sending DELETE for ESP CHILD_SA with SPI c9c89946
Jul 11 00:08:22 prod-nginx charon-custom: 06[IKE] CHILD_SA closed
Jul 11 00:08:22 prod-nginx charon-custom: 06[ENC] generating INFORMATIONAL response 0 [ D ]
Jul 11 00:08:22 prod-nginx charon-custom: 06[NET] sending packet: from 192.*.*.*[4500] to 20.*.*.*[4500] (76 bytes)
Jul 11 00:08:22 prod-nginx ipsec[675]: 10[NET] received packet: from 20.*.*.*[4500] to 192.*.*.*[4500] (76 bytes)
Jul 11 00:08:22 prod-nginx ipsec[675]: 10[ENC] parsed INFORMATIONAL request 1 [ D ]
Jul 11 00:08:22 prod-nginx ipsec[675]: 10[IKE] received DELETE for IKE_SA linode-to-azure[188]
Jul 11 00:08:22 prod-nginx ipsec[675]: 10[IKE] deleting IKE_SA linode-to-azure[188] between 192.*.*.*[172.232.14.79]...20.*.*.*[20.*.*.*]
Jul 11 00:08:22 prod-nginx ipsec[675]: 10[IKE] IKE_SA linode-to-azure[188] state change: ESTABLISHED => DELETING
Jul 11 00:08:22 prod-nginx ipsec[675]: 10[IKE] IKE_SA deleted
Jul 11 00:08:22 prod-nginx ipsec[675]: 10[ENC] generating INFORMATIONAL response 1 [ ]
Jul 11 00:08:22 prod-nginx charon-custom: 10[NET] received packet: from 20.*.*.*[4500] to 192.*.*.*[4500] (76 bytes)
Jul 11 00:08:22 prod-nginx ipsec[675]: 10[NET] sending packet: from 192.*.*.*[4500] to 20.*.*.*[4500] (76 bytes)
Jul 11 00:08:22 prod-nginx ipsec[675]: 10[IKE] IKE_SA linode-to-azure[188] state change: DELETING => DESTROYING
Jul 11 00:08:22 prod-nginx charon-custom: 10[ENC] parsed INFORMATIONAL request 1 [ D ]
Jul 11 00:08:22 prod-nginx charon-custom: 10[IKE] received DELETE for IKE_SA linode-to-azure[188]
Jul 11 00:08:22 prod-nginx charon-custom: 10[IKE] deleting IKE_SA linode-to-azure[188] between 192.*.*.*[172.232.14.79]...20.*.*.*[20.*.*.*]
Jul 11 00:08:22 prod-nginx charon-custom: 10[IKE] IKE_SA deleted
Jul 11 00:08:22 prod-nginx charon-custom: 10[ENC] generating INFORMATIONAL response 1 [ ]
Jul 11 00:08:22 prod-nginx charon-custom: 10[NET] sending packet: from 192.*.*.*[4500] to 20.*.*.*[4500] (76 bytes)

**Reply by tobiasbrunner** [reply]
Just for the record, those log levels are too high (use the ones given in #196 for debugging). And somehow duplicate log messages are produced (maybe due to the syslog config).

> Jul 11 00:08:22 prod-nginx charon-custom: 06[IKE] received DELETE for ESP CHILD_SA with SPI 3462e6c4
> ...
> Jul 11 00:08:22 prod-nginx ipsec[675]: 10[IKE] received DELETE for IKE_SA linode-to-azure[188]

There you go. The other peer explicitly sends deletes for the CHILD_SA and then the IKE_SA. Why? No idea. Ask the other admins for an explanation.

**Reply by harsh-madhav** [reply]
Sure thanks!
